server/chess.py

```diff
@@ -321,8 +321,8 @@ def get_king_moves(self, king: models.Piece) -> typing.Iterator[int, int]:
                 victim = self.get_piece(rank, file)
                 if (not victim) or (victim.side != king.side):
                     yield rank, file
-        for _rook, king_rank in self.get_allowed_castling(king):
-            yield king_rank, king.file
+        for _rook, king_file in self.get_allowed_castling(king):
+            yield king.rank, king_file
```

Comment:
Good catch, I mixed up ranks and files when writing this and had to go through and change about 300, must have missed this.